src/components/Workspace.jsx

```diff
@@ -65,13 +67,27 @@ function mapStateToProps(state) {
     errors: state.get('errors').toJS(),
     runtimeErrors: state.get('runtimeErrors').toJS(),
     isUserTyping: state.getIn(['ui', 'editors', 'typing']),
-    editorsFlex: state.getIn(['ui', 'editors', 'verticalFlex']).toJS(),
+    editorsFlex: state.getIn(['ui', 'workspace', 'columnFlex']).toJS(),
+    rowsFlex: state.getIn(['ui', 'workspace', 'rowFlex']).toJS(),
     currentUser: state.get('user').toJS(),
     ui: state.get('ui').toJS(),
     clients: state.get('clients').toJS(),
   };
 }
 
+function getNodeWidth(node) {
+  const minWidth = window.getComputedStyle(node).minWidth;
+  return {
+    width: node.offsetWidth,
+    minWidth: parseInt(minWidth.replace('px', ''), 10),
+  };
+}
+
+function getNodeWidths(refs) {
+  return Array.from(refs).sort().map(([, node]) =>
```

Comment:
I think the fact that the component stores the `refs` in a map is outside the concern of this function—I’d `map` them before passing them into this function.